File: core/src/main/java/io/opentelemetry/android/NoopOpenTelemetryRum.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android;

import io.opentelemetry.api.OpenTelemetry;
import io.opentelemetry.api.common.Attributes;
import io.opentelemetry.api.trace.TraceId;

enum NoopOpenTelemetryRum implements OpenTelemetryRum {
    INSTANCE;

    @Override
    public OpenTelemetry getOpenTelemetry() {
        return OpenTelemetry.noop();
    }

    @Override
    public String getRumSessionId() {
        // RUM session.id has the same format as traceId
        return TraceId.getInvalid();
    }

    @Override
    public void emitEvent(String eventName, String body, Attributes attributes) {}

    @Override
    public void shutdown() {
        // nop
    }
}

File: core/src/main/java/io/opentelemetry/android/NoopOpenTelemetryRum.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android

import io.opentelemetry.android.session.SessionProvider
import io.opentelemetry.api.OpenTelemetry
import io.opentelemetry.api.common.Attributes
import io.opentelemetry.sdk.common.Clock

internal object NoopOpenTelemetryRum : OpenTelemetryRum {
    override val openTelemetry: OpenTelemetry = OpenTelemetry.noop()
    override val sessionProvider: SessionProvider = SessionProvider.getNoop()
    override val clock: Clock = Clock.getDefault()

    override fun emitEvent(
        eventName: String,
        body: String,
        attributes: Attributes,
    ) {
    }

    override fun shutdown() {
    }
}
